File: examples/node/large-file-async.mjs
// Large or multi-page documents: use an async job (up to 50 MB and 500 pages).
// Poll with job.wait(), or pass webhookUrl to be notified.
//
//   node large-file-async.mjs statement_2025.pdf statement_2025.xlsx
import { MyOCRClient } from 'myocr-client';

const [src = '../sample_bank_statement.pdf', dst = 'statement.xlsx'] = process.argv.slice(2);
const client = new MyOCRClient();

const job = await client.createJob(src, { model: 'bank_statement' });
console.log(`Job ${job.requestId} queued`);

await job.wait({ timeoutMs: 15 * 60 * 1000 }); // polls with exponential backoff
if (job.isDone) {
  await job.download(dst);
  console.log(`Saved ${dst}`);
} else {
  console.log(`Job ended with status ${job.status}: ${job.errorDetail}`);
  process.exitCode = 1;
}

File: examples/python/large_file_async.py
"""Large or multi-page documents: use an async job.

/v1/convert is synchronous and takes files up to 5 MB and 10 pages.
Jobs take up to 50 MB and 500 pages. Poll with job.wait(), or pass
webhook_url=... to be notified (see webhook_receiver.py).

    python large_file_async.py statement_2025.pdf statement_2025.xlsx
"""
import sys

from myocr_client import MyOCRClient

src = sys.argv[1] if len(sys.argv) > 1 else "../sample_bank_statement.pdf"
dst = sys.argv[2] if len(sys.argv) > 2 else "statement.xlsx"

client = MyOCRClient()
job = client.create_job(src, model="bank_statement")
print(f"Job {job.request_id} queued")

job.wait(timeout=900)  # polls with exponential backoff
if job.is_done:
    job.download(dst)
    print(f"Saved {dst}")
else:
    print(f"Job ended with status {job.status}: {job.error_detail}")
    sys.exit(1)
